### ISSUE TITLE
Cert-issuer Ethereum testnet setup failing

### ISSUE THREAD

**Issue by endlos** [blockchain-certificates/cert-issuer#107]
I followed the steps [here](https://github.com/blockchain-certificates/cert-issuer#getting-the-docker-image) to build the docker container for **cert-issuer**, and also replacing libressl-dev for openssl-dev to overcome the errors I got, as shown [here](https://github.com/blockchain-certificates/cert-issuer/issues/100#issuecomment-392952918).

Then after  running this (using python3.6 with virtualenv):

    (venv) bash-4.3# python setup.py experimental --blockchain=ethereum

I got this error:

    ethereum 2.3.2 has requirement rlp<2.0.0,>=1.0.1, but you'll have rlp 0.6.0 which is incompatible.
    Installing collected packages: rlp
      Found existing installation: rlp 1.0.2
        Uninstalling rlp-1.0.2:
          Successfully uninstalled rlp-1.0.2
    Successfully installed rlp-0.6.0

I found [this](https://community.blockcerts.org/t/error-while-installing-blockcerts-for-use-with-ethereum/1008/5) which suggests adding **rlp<1** to "ethereum_requirements.txt", but it is already added, so I changed "**rlp<1**" to "**rlp>=1**" and the setup seemed to complete successfully. 

Then I get this error, after running:

    (venv) bash-4.3# cert-issuer -c /etc/cert-issuer/conf.ini
...

    Traceback (most recent call last):
      File "/usr/bin/cert-issuer", line 11, in <module>
        load_entry_point('cert-issuer==2.0.15', 'console_scripts', 'cert-issuer')()
      File "/usr/lib/python3.6/site-packages/cert_issuer/__main__.py", line 17, in cert_issuer_main
        issue_certificates.main(parsed_config)
      File "/usr/lib/python3.6/site-packages/cert_issuer/issue_certificates.py", line 36, in main
        return issue(app_config, certificate_batch_handler, transaction_handler)
      File "/usr/lib/python3.6/site-packages/cert_issuer/issue_certificates.py", line 22, in issue
        tx_id = issuer.issue(app_config.chain)
      File "/usr/lib/python3.6/site-packages/cert_issuer/issuer.py", line 27, in issue
        txid = self.transaction_handler.issue_transaction(blockchain_bytes)
      File "/usr/lib/python3.6/site-packages/cert_issuer/blockchain_handlers/ethereum/transaction_handlers.py", line 61, in issue_transaction
        self.verify_transaction(signed_tx, eth_data_field)
      File "/usr/lib/python3.6/site-packages/cert_issuer/blockchain_handlers/ethereum/transaction_handlers.py", line 92, in verify_transaction
        tx_utils.verify_eth_transaction(signed_tx, eth_data_field)
      File "/usr/lib/python3.6/site-packages/cert_issuer/blockchain_handlers/ethereum/tx_utils.py", line 24, in verify_eth_transaction
        for s in signed_hextx.split('80a0'):
      AttributeError: 'dict' object has no attribute 'split'

Previously reported this in the Community forum in [this post](https://community.blockcerts.org/t/cert-issuer-ethereum-testnet-setup-failing/1316) and posting it here as requested.

Thanks.

**Comment by stuartf** [comment]
I'm seeing this as well. It looks like `verify_eth_transaction` is expecting a string that it can split, but gets a dict representing an error: `{'error': True, 'message': AttributeError("can't set attribute",)}` that came out of `sign_transaction`

**Comment by endlos** [comment]
A workaround is forcing ethereum==2.3.1 while leaving rlp<1 in ethereum_requirements.txt
```
ethereum==2.3.1
rlp<1
```
Credits go to @AnthonyRonning for his suggestion in the forum [here]!(https://community.blockcerts.org/t/cert-issuer-ethereum-testnet-setup-failing/1316/8?u=hugo)

**Comment by endlos** [comment]
A workaround is forcing ethereum==2.3.1 while leaving rlp<1 in ethereum_requirements.txt
```
ethereum==2.3.1
rlp<1
```
Credits go to @AnthonyRonning for his suggestion in the forum [here](https://community.blockcerts.org/t/cert-issuer-ethereum-testnet-setup-failing/1316/8?u=hugo)!

**Comment by endlos** [comment]
A workaround is forcing ethereum==2.3.1 while leaving rlp<1 in ethereum_requirements.txt
```
ethereum==2.3.1
rlp<1
```
Credits go to @AnthonyRonning for his suggestion in the forum [here](https://community.blockcerts.org/t/cert-issuer-ethereum-testnet-setup-failing/1316/8?u=hugo)!

**Comment by stuartf** [comment]
I'd suggest leaving this open or opening a new issue that those dependency versions need to be pinned or the code updated to work with the new versions.

**Comment by AnthonyRonning** [comment]
Thanks @stuartf, I agree. I'll get a PR for that fix. Thanks for confirming that @endlos

**Comment by AnthonyRonning** [comment]
We've got that merged in, and opened #110 tracking the dependency issue in the first place. Always good to be on the latest versions, we just need to ensure that they work.
